core/commands/refs.go

```diff
@@ -86,6 +89,12 @@ Note: List all referencess recursively by using the flag '-r'.
 			return
 		}
 
+		cached, _, err := req.Option("cached").Bool()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
```

Comment:
now that you have that, you don't need any of the `Cached` logic in the RefWriter